### PR TITLE
16.0 mig base location nuts

### PR DESCRIPTION
**Pull request by Vincenzo9191**
Migration to version 16.0 for base_location_nuts (issue #1353 )
- Changed library for HTTP GET from requests to urllib3;
- Changed test class to TestBaseLocationNuts(common.TransactionCase) because SavepointCase has been merged in TransactionCase;

### CODE REVIEW

File: base_location_nuts/wizard/nuts_import.py [hunk 1]
```diff
@@ -121,7 +121,7 @@ def _mapping(self, node):
             else:
                 logger.debug("xpath = '%s', not found" % field_xpath)
             if field_required and not value:
-                raise UserError(_("Value not found for mandatory field %s" % k))
+                raise UserError(_("Value not found for mandatory field %s") % k)
```
Comment (tafaRU) [inline base_location_nuts/wizard/nuts_import.py:124]: Good catch!

File: base_location_nuts/wizard/nuts_import.py [hunk 2]
```diff
@@ -136,20 +136,24 @@ def _download_nuts(self, url_base=None, url_path=None, url_params=None):
...
         try:
-            res_request = requests.get(url)
+            http = urllib3.PoolManager()
+            res_request = http.request("GET", url)
         except Exception as e:
             raise UserError(
                 _("Got an error when trying to download the file: %s.") % str(e)
-            )
-        if res_request.status_code != requests.codes.ok:
+            ) from e
```
Comment (tafaRU) [inline base_location_nuts/wizard/nuts_import.py:144]: Is this change needed?

**Review by tafaRU — CHANGES_REQUESTED** [review]
Thanks for your PR!
Please squash administrative commits for reducing commit noise according to https://github.com/OCA/maintainer-tools/wiki/Merge-commits-in-pull-requests#mergesquash-the-commits-generated-by-bots-or-weblate

The rest LGTM, also tested :+1:

File: base_location_nuts/wizard/nuts_import.py (hunk 2 above)
Comment (Vincenzo9191) [reply base_location_nuts/wizard/nuts_import.py:144]: Yes, it is a change due to the good presentation of the error, otherwise it raised a warning.

**Comment by tafaRU** [comment]
@Vincenzo9191 there is still https://github.com/OCA/partner-contact/pull/1398#pullrequestreview-1194567899 to address. Could you please take a look?

**Comment by Vincenzo9191** [comment]
> @Vincenzo9191 there is still [#1398 (review)](https://github.com/OCA/partner-contact/pull/1398#pullrequestreview-1194567899) to address. Could you please take a look?

@tafaRU thank you for your review;
If may ask, can you address me the initial and last commit to squash to follow the comunity guidelines?

**Comment by tafaRU** [comment]
@Vincenzo9191 
the first one is the following:

image

while the last one is:

image

In the middle the others!

Please note that in case of multiple administrative commits as the following:

image

you have to squash all the commits to only one, the real one!

**Comment by tafaRU** [comment]
@Vincenzo9191 @FrancescoCosma there are still administrative commits need to be squashed:

16-0-mig-base-location-nuts-by-Vincenzo9191-·-Pull-Request-1398-·-OCA-partner-contact

Could you also please squash these latter?

**Review by tafaRU — APPROVED** [review]

**Comment by primes2h** [comment]
@Vincenzo9191 @FrancescoCosma 

Please make a fixup combining these two commits in one too. Aside from that, LGTM.

immagine

**Comment by tafaRU** [comment]
@Vincenzo9191 @FrancescoCosma could you please address the last comment? So we can proceed with the merge.
Thank you in advance.

**Comment by tafaRU** [comment]
@primes2h could you please update your review?

**Comment by primes2h** [comment]
@Vincenzo9191 @FrancescoCosma 
Runboat is missing, please force a push to create it.

**Comment by tafaRU** [comment]
@Vincenzo9191 please squash last two commits to only one as the latter is not significant. Thanks!

**Comment by tafaRU** [comment]
@FrancescoCosma could you please have a look at https://github.com/OCA/partner-contact/pull/1398#issuecomment-1537859273?

**Review by EmilioPascual — CHANGES_REQUESTED** [review]
Code Review OK
Funcional review: I found some problems. I can't download nuts file.
Captura desde 2023-06-22 11-19-12
Captura desde 2023-06-22 11-19-38
Captura desde 2023-06-22 11-19-49

**Comment by matteoopenf** [comment]
> Code Review OK Funcional review: I found some problems. I can't download nuts file. Captura desde 2023-06-22 11-19-12 Captura desde 2023-06-22 11-19-38 Captura desde 2023-06-22 11-19-49

the same problem there is in 14.0

**Comment by thomaspaulb** [comment]
Peeps, the RAMON server seems to have gone offline, and so the download cannot work anymore.

It seems like [this](https://circabc.europa.eu/ui/group/c1b49c83-24a7-4ff2-951c-621ac0a89fd8/library/579984e5-610d-4adb-af35-a16e014a3cd9) has some archived data where we might be able to find the XML, or, we have to refactor in order to use the new [sparql endpoint](https://ec.europa.eu/eurostat/web/metadata/classifications) to the data.

**Comment by pedrobaeza** [comment]
The second one seems the correct way to go.

**Comment by NICO-SOLUTIONS** [comment]
> The second one seems the correct way to go.

Anybody attacking this issue already?

**Comment by matteoopenf** [comment]
> > The second one seems the correct way to go.
> 
> 
> 
> Anybody attacking this issue already?

Not for now

**Comment by rafaelbn** [comment]
@Vincenzo9191 could you please take care about the XML in the commenst above is the correct one 😄 !
